### PR TITLE
Improve messages workflow

### PR DESCRIPTION
**Pull request by misstick**
## Improve features
 - [x] upgrade `unread messages counter` into menu as fast as possible,
 - [x] active realtime events,
 - [x] fix `nextURL` setting,
 - [x] do not fetch all messages to get `conversationLength`,
 - [x] fetch all messages from conversation when a message is displayed,
 - [x] show only current message on 2nd panel,
 - [x] reset `currentID` when Route changes,
 - [x] clean messageList dead code,
 - [x] fix keyboard navigation for `top` and `bottom` keys,
 - [x] fix `filter` navigation,
 - [x] fix checkbox `selection`,

<strike> - [ ] `selection` menu: remove tools submenu to display simple buttons </strike>

## Remove features
 - [x] `search bar` is useless (search is broken),
 - [x] `filter toolbar`,
 - [x] add mailbox `unread`,
 - [x] add mailbox `important`,
 - [x] compose was re-added but should disappear, see: https://github.com/cozy/cozy-emails/pull/807

### CODE REVIEW

File: client/app/components/conversation.coffee [hunk 1]
```diff
@@ -106,7 +103,7 @@ module.exports = React.createClass
 
             section
                 ref: 'scrollable',
-                    @state.messages.map @renderMessage
+                    @state.conversation.map @renderMessage
 
     _initScroll: ->
         if not (scrollable = ReactDOM.findDOMNode @refs.scrollable) or scrollable.scrollTop
```
Comment (frankrousseau) [inline client/app/components/conversation.coffee]: Are you sure that this line is < 80 chars?

**Comment by frankrousseau** [comment]
I didn't notice before but it looks like you don't follow the 80 chars guideline. Could you stick to it? It makes the text size standard, force to write more concise code and match with what we do since the beginning.

**Comment by frankrousseau** [comment]
Too much PR open for the end of the sprint. Please merge or close them on monday.

**Comment by misstick** [comment]
@m4dz : I've juste finished the fiw because of merge; you can review this PR when ever you want :]

File: client/app/components/search_bar.coffee [hunk 2]
```diff
@@ -66,7 +66,7 @@ module.exports = GlobalSearchBar = React.createClass
         .toOrderedMap()
         .set 'all', t 'search all accounts'
 
-        accountID = AccountStore.getSelected()?.get('id') or 'all'
+        accountID = AccountStore.getAccountID() or 'all'
```
Comment (m4dz) [inline client/app/components/search_bar.coffee]: Should use a getter here, nope?

File: client/app/stores/account_store.coffee [hunk 3]
```diff
@@ -379,62 +252,9 @@ class AccountStore extends Store
...
-        account.imapTLS = false
-        account.accountType = 'IMAP'
-        account.favoriteMailboxes = null
-        return Immutable.Map account
-
-
 
 
-module.exports = new AccountStore()
+module.exports = _self = new AccountStore()
```
Comment (m4dz) [inline client/app/stores/account_store.coffee]: Why storing instance in `_self`?

**Comment by m4dz** [comment]
I merge as it, maybe we need to think about:
- using Getter in component and never call Stores directly
- Use dedicated getters for stores like MessagesStores, and not let RouterGetter serves contents that doesn't rely directly on it (like `getMessagesList`, which should be in a MessagesGetter that can ask RouterSTore for the current `mailboxID`)

We also need to add a lot of comments in the whole app: each time I review, it's deeply hard to understand who does what, how, and what have changed :disappointed:…

**Comment by misstick** [comment]
@m4dz : 
 - " using Getter in component and never call Stores directly" : this should be done a little each time 
 - for getter, we could talk about this; to my mind, having only one getter si enought but its name should be change in `getters/application` ie.

File: client/app/stores/account_store.coffee (hunk 3 above)
Comment (misstick) [inline client/app/stores/account_store.coffee]: Sometime references to `this` with `@` doesnt work...

**Pull request merged**